### PR TITLE
Fix `show-open-prs-of-forks` to only include related PRs 

### LINKED ISSUE #2934: `show-open-prs-of-forks` includes unrelated PRs
1. Visit one of your (source) repos where **you** opened PRs
2. Fork it
3. Open it
4. Notice: your existing PRs are counted in this feature

In my case, I have this dummy fork https://github.com/fregante/refined-github but all of PRs are directly on the source (https://github.com/sindresorhus/refined-github)

cc @loilo

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   Fixes #2934 I also fixed a console error while I was at it. Follows https://github.com/sindresorhus/refined-github/issues/2934#issuecomment-604146813 (I changed it to headRepository because I think that is correct)

2. TEST URLS:
   I tested on https://github.com/talknsmile/refined-github which is a fork of my fork and I have 3 local PR's open and I have only one pull request

3. SCREENSHOT:
   Before
image

	After 
image

### CODE REVIEW

File: source/features/show-open-prs-of-forks.tsx [hunk 1]
```diff
@@ -6,35 +6,38 @@ import * as pageDetect from 'github-url-detection';
...
 				}
 			}
 		}
 	`);
 
-	if (search.issueCount === 1) {
-		return [1, search.nodes[0].number];
+	const prs = search.nodes.filter((pr: AnyObject) => pr.headRepository.nameWithOwner.toLowerCase() === getRepoURL());
+
+	// If only one is found, pass the pr number so we can link to the PR directly
```
Comment (fregante) [inline source/features/show-open-prs-of-forks.tsx:35]:
```suggestion
	// If only one is found, pass the PR number so we can link to the PR directly
```

File: source/features/show-open-prs-of-forks.tsx [hunk 2]
```diff
@@ -6,35 +6,38 @@ import * as pageDetect from 'github-url-detection';
...
+						nameWithOwner
+					}
 				}
 			}
 		}
 	`);
 
-	if (search.issueCount === 1) {
-		return [1, search.nodes[0].number];
+	const prs = search.nodes.filter((pr: AnyObject) => pr.headRepository.nameWithOwner.toLowerCase() === getRepoURL());
```
Comment (fregante) [inline source/features/show-open-prs-of-forks.tsx:33]:
```suggestion
	// Only show PRs originated from the current repo
	const prs = search.nodes.filter((pr: AnyObject) => pr.headRepository.nameWithOwner.toLowerCase() === getRepoURL());
```

File: source/features/show-open-prs-of-forks.tsx [hunk 3]
```diff
@@ -69,7 +72,7 @@ async function initHeadHint(): Promise<void | false> {
 
 async function initDeleteHint(): Promise<void | false> {
 	const [count, url] = await getPRs();
-	if (!count) {
+	if (!count || !location.pathname.endsWith('settings')) {
```
Comment (fregante) [inline source/features/show-open-prs-of-forks.tsx:75]: This needs to be more precise (this also matches `/fregante/webext-settings`) and belongs to the `exclude`. It also already exists: https://github.com/fregante/github-url-detection/blob/875cf8a5e67bb1332290f52dc1f7a8ec0c50d3e1/index.ts#L299
```js

	exclude: [
		() => !pageDetect.isForkedRepo(),
		pageDetect.isRepoSettings
	],
```

**Comment by Hornet171717** [comment]
I already filled a complaint with the federal trade commission since I seen the good Dr was worried about that and I’m not done yet

File: source/features/show-open-prs-of-forks.tsx (hunk 3 above)
Comment (yakov116) [reply source/features/show-open-prs-of-forks.tsx]: The error is when going to the setting, we only want to run it on the page the has the delete option.
In other words we are in the settings.

Correct me if I am wrong.

I moved it to exclude in f5f35edc1e185c364f5650bbe37ef9ec5968420b

Comment (fregante) [reply source/features/show-open-prs-of-forks.tsx]: You can use https://github.com/fregante/github-url-detection/pull/2

Comment (yakov116) [reply source/features/show-open-prs-of-forks.tsx]: Done

**Comment by fregante** [comment]
Thank you!

**Pull request merged**